profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoopManager.cpp

```diff
@@ -519,8 +518,7 @@ void StackSamplerLoopManager::NotifyIterationFinished()
     std::lock_guard<std::mutex> guardedLock(_watcherActivityLock);
 
     _pTargetThread->GetStackWalkLock().Release();
-    _pTargetThread->Release();
-    _pTargetThread = nullptr;
+    _pTargetThread.reset();
```

Comment:
Is this what will make the nullptr test in line 393 work?
`if (LogDuringStackSampling_Unsafe || _pTargetThread == nullptr)`

Reply:
exactly.

Reply:
Calling reset will call Release(). FYI